packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryResetPassword/index.tsx

```diff
@@ -314,18 +314,20 @@ const AccountRecoveryResetPassword = ({
     setBannerState(BannerState.PasswordResetSuccess);
   }
 
-  function navigateAway() {
-    setUserPreference('account-recovery', account.recoveryKey);
+  async function navigateAway() {
+    // TODO / refactor: the initial account result with useAccount() does not
+    // contain account data due to no session token. Users receive the session
+    // token on PW reset, so we can query here for it.
+    const hasTotp = await account.hasTotpAuthClient();
```

Comment:
Just realized, this should be in a `try`/`catch` here or in `Account.ts`, I'll add that before merge

Reply:
Done. Also added [this check](https://github.com/mozilla/fxa/pull/15392/files#diff-99ad6a399825850b7a5cdb423368271e810e574014e2a70c35864c2653d0496fR236) in case the check failed, where I'm just returning `false` as a fallback, but if this is `false` when the user does have TOTP, they’d get stuck in a weird state where they’d see a flash of reset PW verified and then they’d be taken to `/signin_token_code` (since that function was getting hit), with an undefined email which I couldn’t get out of without hitting `/clear` since it was hitting that `mustVerify` issue.